app/src/main/java/com/babylon/wallet/android/presentation/onboarding/restore/backup/CloudBackupLoginScreen.kt

```diff
@@ -0,0 +1,190 @@
+    onContinueToRestoreFromBackup: () -> Unit
+) {
+    val state by viewModel.state.collectAsStateWithLifecycle()
+
+    BackHandler { onBackClick() }
+
+    val signInLauncher = rememberLauncherForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+        viewModel.handleSignInResult(result)
+    }
+
+    val recoverDriveAuthLauncher = rememberLauncherForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+        viewModel.handleAuthDriveResult(result)
+    }
+
+    LaunchedEffect(Unit) {
+        viewModel.oneOffEvent
+            .collect { event ->
+                when (event) {
+                    is CloudBackupLoginViewModel.Event.SignInToGoogle -> {
+                        signInLauncher.launch(event.signInIntent)
+                    }
+
+                    is CloudBackupLoginViewModel.Event.RecoverUserAuthToDrive -> {
+                        recoverDriveAuthLauncher.launch(event.authIntent)
+                    }
+
+                    is CloudBackupLoginViewModel.Event.ProceedToRestoreFromBackup -> {
+                        onContinueToRestoreFromBackup() // navigate to RestoreFromBackupScreen
+                    }
+                }
+            }
+    }
+
+    CloudBackupLoginContent(
+        modifier = modifier,
+        isAccessToGoogleDriveInProgress = state.isAccessToGoogleDriveInProgress,
+        errorMessage = state.errorMessage,
+        onErrorMessageShown = viewModel::onErrorMessageShown,
+        onBackClick = onBackClick,
+        onLoginToGoogleClick = viewModel::onLoginToGoogleClick,
+        onSkipClick = viewModel::onSkipClick
+    )
+}
+
+@Composable
+private fun CloudBackupLoginContent(
+    modifier: Modifier = Modifier,
+    isAccessToGoogleDriveInProgress: Boolean,
+    errorMessage: UiMessage?,
+    onErrorMessageShown: () -> Unit,
+    onBackClick: () -> Unit,
+    onLoginToGoogleClick: () -> Unit,
+    onSkipClick: () -> Unit
+) {
+    val snackBarHostState = remember { SnackbarHostState() }
+    SnackbarUIMessage(
+        message = errorMessage,
+        snackbarHostState = snackBarHostState,
+        onMessageShown = onErrorMessageShown
+    )
+
+    Scaffold(
+        modifier = modifier.imePadding(),
+        topBar = {
+            RadixCenteredTopAppBar(
+                title = "",
+                onBackClick = onBackClick,
+                actions = {
+                    RadixTextButton(
+                        text = "Skip", // TODO Crowdin
+                        onClick = onSkipClick
+                    )
+                },
+                windowInsets = WindowInsets.statusBars,
+            )
+        },
+        snackbarHost = {
+            RadixSnackbarHost(
+                modifier = Modifier.padding(RadixTheme.dimensions.paddingDefault),
+                hostState = snackBarHostState
+            )
+        },
+        containerColor = RadixTheme.colors.defaultBackground
+    ) { padding ->
+
+        Column(
```

Comment:
this composable is almost identical as in `ConnectCloudBackupScreen`, just different copies and onClick handler. Maybe you can make content of those screens parametrized composable?

Reply:
yup I think we can do this.